### PR TITLE
Device description in dashboard

### LINKED ISSUE #377: Feature Request: Device Documentation in Dashboard
**Describe the problem you have/What new integration you would like**
I really like the ESPHome Dashboard but I'm missing one functionality.
I would like to see the actual entity names or functionality there since it is not always clear from my device names.
(Example Sonoff 4 Ch providing 4 different controls. If you have 10 of them it's a mess to remember which channel is providing which signal or an external documentation is needed.)

**Please describe your use case for this integration and alternatives you've tried:**
Two possibilities to implement it:
1. **is my favorite:** just include a "Comment" Field in "esphome:" section which can be used to add custom text / html to the overview. This gives huge degree of freedom for all kind of documentation without breaking the approach to keep the dashboard as simple as possible
2. automatically determine all entity names provided by a device and show them beside the device name. I think this is to specific and complicated as you already commented for #132 .

**Additional context**

### PR DESCRIPTION
**Pull request by Anonym-tsk**
## Description:
Ability to add additional descriptions to devices in config and dashboard.

**Related issue (if applicable):** fixes https://github.com/esphome/feature-requests/issues/377, https://github.com/esphome/feature-requests/issues/353

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#335

desc

```yaml
esphome:
  name: my_mega_device
  comment: "Light in kitchen with motion sensor"
  platform: ESP8266
  board: esp01_1m
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

@OttoWinter what do you think about this feature and my code?

### CODE REVIEW

File: esphome/wizard.py [hunk 1]
```diff
@@ -47,6 +47,7 @@
 
 BASE_CONFIG = u"""esphome:
   name: {name}
+  description: "{description}"
```
Comment (OttoWinter) [inline esphome/wizard.py]: Let's not add this to the wizard, I would consider this an advanced feature. The setup wizard should only include the bare-minimum for creating a new config, not nice-to-have extras.

File: esphome/core_config.py [hunk 2]
```diff
@@ -164,6 +166,7 @@ def preload_core_config(config):
     with cv.prepend_path(core_key):
         out = PRELOAD_CONFIG_SCHEMA(config[CONF_ESPHOME])
     CORE.name = out[CONF_NAME]
+    CORE.description = out.get(CONF_DESCRIPTION)
```
Comment (OttoWinter) [inline esphome/core_config.py]: This should not be set in here, instead use a `@property` in the CORE object to automatically fetch the description from the config.

**Review by OttoWinter — COMMENTED** [review]
Sounds like an interesting feature 👍

File: esphome/wizard.py (hunk 1 above)
Comment (Anonym-tsk) [reply esphome/wizard.py]: I think, it's very good to add description immediately on create new node.
This field is optional in wizard.

Comment (OttoWinter) [reply esphome/wizard.py]: Don't get me wrong, I think this is great for people with lots of devices around their house, but I doubt the average *new* user (towards whom the setup wizard is geared towards) will know what to do with this field.

What do you think a new user (potentially with a test device and not knowing what exactly they want to build yet) will input here? Setup wizard should be bare minimum, for example we don't ask the user about the log level here either.

If the user wants to use an optional feature that is not necessary for basic functioning, they can edit the config. The setup wizard should take up the least amount of your time as possible.

> This field is optional in wizard.

In my experience, its the existence of the field - I haven't looked at how it looks in the wizard UI yet, but if it's just another field I doubt the user will understand that it's optional.

Comment (Anonym-tsk) [reply esphome/wizard.py]: ESPHome Dashboard 2019-08-29 17-33-40
I can mark this field as `Description (optional)`

Comment (Anonym-tsk) [reply esphome/wizard.py]: @OttoWinter I would like to add this field to wizard.
But, of course, last word is yours.

Comment (OttoWinter) [reply esphome/wizard.py]: See, I would still appreciate an argument why this field would be *necessary* for the setup wizard. Yes I agree it's a handy feature but is it really necessary for the wizard? I'd be happy to accept it if I get a convincing argument.

As for the design: `Description (optional)` would definitely be better, but there would need to be some explanation of what description really means (and what it effects). Sure, I understand what it means, but think about the new user getting shown a field with no explanation of what it means (and the potential semantics of said field). But then the whole first page (which is already too long) would get an even longer wall of text.

**Comment by OttoWinter** [comment]
Another thing we could think about: https://github.com/esphome/feature-requests/issues/377 gave me an idea. We could change `description` to `comment` - That would emphasize that it doesn't have any *semantical* meaning, and is just for the user. Idk, probably not that important but worth thinking about.

File: esphome/wizard.py (hunk 1 above)
Comment (Anonym-tsk) [reply esphome/wizard.py]: So, i renamed `description` to `comment` and removed it from wizard.

**Comment by Anonym-tsk** [comment]
So, i renamed `description` to `comment` and removed it from wizard.

**Review by OttoWinter — APPROVED** [review]

**Pull request merged**